lib/s3_meta_sync/syncer.rb

```diff
@@ -83,6 +87,23 @@ def download(source, destination)
       end
     end
 
+    # Sometimes SIGTERM causes Dir.mktmpdir to not properly clear the temp folder
+    # Remove day old folders
+    def clear_old_temp_folders
+      path = File.join(Dir.tmpdir, TEMP_FOLDER_PREFIX + '*')
+
+      total = Dir.glob(path).map do |dir|
+        day = 60 * 60 * 24
+        FileUtils.rm_rf(dir) if older_than(dir, day)
+      end.compact.count
+
+      log "Removed #{total} old temp folder(s)" if total > 0
```

Comment:
might be interesting to say which folders were removed since that indicates a previous error ... and could be useful to search the logs for what failed